packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -998,7 +1001,32 @@ class HeroController extends NavigatorObserver {
     BuildContext toHeroContext,
   ) {
     final Hero toHero = toHeroContext.widget as Hero;
-    return toHero.child;
+    if (MediaQuery.maybeOf(toHeroContext) == null && MediaQuery.maybeOf(fromHeroContext) == null) {
```

Comment:
I'm leaning towards only doing this when both `MediaQuery` are not null. In theory there could be cases where the Hero subtree has a hard dependency on MediaQuery and should throw when there's no MediaQuery present in the tree. Adding a `MediaQuery` in this case would prevent the developer from getting the error.

In a real app you'll always have a `MediaQuery` if you use `WidgetsApp` so it doesn't really matter. But that could make the interpolation logic slightly easier to write.

Reply:
I agree. Defaulting empty `MediaQueryData` to the other can lead to confusing result.

Imagine there are 3 trees, A, B, and C. A and C has non-empty `MediaQueryData`, while B has none. A hero X flies from A to B, then from B to C. During the first flight, it uses a fixed padding from A, while during the second flight, it uses a fixed padding from C. This is a jumping animation. 

We might want to either default empty paddings to zero, or assert this situation.

Reply:
I believe just not doing a transition if either MediaQuery is wrong fixes the possibility of this issue, though correct me if I'm wrong.

Reply:
I'm leaning to asserting it. Reason:
- This is rare. The hero is transitioning from within a MediaQuery to out of it. The developer is doing something advanced but is doing it wrong, and it might be better to tell them.
- If we don't do transitions at all, this almost certainly leads to a jumping animation, which is another issue to be raised, investigated, and fixed in the future.